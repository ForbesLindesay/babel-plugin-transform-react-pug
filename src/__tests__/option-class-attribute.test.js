import testHelper from './test-helper';

const consoleError = console.error.bind(console);
beforeAll(() => (console.error = jest.fn()));
afterAll(() => (console.error = consoleError));

testHelper(__dirname + '/option-class-attribute.input.js', {
  classAttribute: 'styleName',
});
