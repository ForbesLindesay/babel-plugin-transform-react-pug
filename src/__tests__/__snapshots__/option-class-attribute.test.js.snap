// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JavaScript output: transformed source code 1`] = `
"const name = 'jack';

module.exports = <div styleName=\\"first-a first-b\\"><p styleName=\\"second-a second-b second-c second-d\\" /><p styleName=\\"third-a third-b third-c third-d third-e third-f third-g third-h\\" /><p styleName=\\"fourth-a fourth-b\\" /><p styleName=\\"fivth-a fivth-b\\" />{((_pug_nodes, _pug_arr) => {
    if (!(_pug_arr == null || Array.isArray(_pug_arr))) throw new Error('Expected \\"[1, 2, 3]\\" to be an array because it is passed to each.');
    if (_pug_arr == null || _pug_arr.length === 0) return undefined;

    for (let _pug_index = 0; _pug_index < _pug_arr.length; _pug_index++) {
      const item = _pug_arr[_pug_index];
      _pug_nodes[_pug_nodes.length] = <p styleName={['seventh-a', 'seventh-b', \\"seventh-i-\\" + item].join(' ')} key={'pug' + item + ':0'} />;
    }

    return _pug_nodes;
  })([], [1, 2, 3])}</div>;"
`;

exports[`html output: generated html 1`] = `
<div
  styleName="first-a first-b"
>
  <p
    styleName="second-a second-b second-c second-d"
  />
  <p
    styleName="third-a third-b third-c third-d third-e third-f third-g third-h"
  />
  <p
    styleName="fourth-a fourth-b"
  />
  <p
    styleName="fivth-a fivth-b"
  />
  <p
    styleName="seventh-a seventh-b seventh-i-1"
  />
  <p
    styleName="seventh-a seventh-b seventh-i-2"
  />
  <p
    styleName="seventh-a seventh-b seventh-i-3"
  />
</div>
`;

exports[`static html output: static html 1`] = `"<div styleName=\\"first-a first-b\\"><p styleName=\\"second-a second-b second-c second-d\\"></p><p styleName=\\"third-a third-b third-c third-d third-e third-f third-g third-h\\"></p><p styleName=\\"fourth-a fourth-b\\"></p><p styleName=\\"fivth-a fivth-b\\"></p><p styleName=\\"seventh-a seventh-b seventh-i-1\\"></p><p styleName=\\"seventh-a seventh-b seventh-i-2\\"></p><p styleName=\\"seventh-a seventh-b seventh-i-3\\"></p></div>"`;
